src/elements/element.rectangle.js

```diff
@@ -50,42 +50,76 @@ module.exports = function(Chart) {
 	}
 
 	Chart.elements.Rectangle = Chart.Element.extend({
-		draw: function() {
+		draw: function(isHorizontalBar) {
```

Comment:
A rectangle should be a primitive defined by `x, y, h, w`, no more (its orientation depends on the h/w ratio). `isHorizontalBar` is chart logic which should be handled in the bar controllers (certainly in `updateElement`). If we want to re-use the rectangle element for a completely different chart type, then it doesn't really make sense to have the bar chart logic here. Two alternative solutions:

1. (easiest) make the name of this argument more generic, maybe just `horizontal` to avoid associating the rectangle API to the bar chart. Also, I don't think it should be a `draw` argument but a property of the `model` (as `vm.x`, `vm.y`, etc.) defined in the controller `updateElement` (`rectangle._model.horizontal = true/false`).
2. (favorite) simplify the rectangle implementation to only work with `vm.x`, `vm.y`, `vm.height` and `vm.width` (so remove  `vm.base`) and let controllers correctly setup these coordinates.

What do you think?